storage_drivers/azure/azure_anf_subvolume.go

```diff
@@ -559,17 +562,36 @@ func (d *NASBlockStorageDriver) initializeAzureSDKClient(
 		}
 	}
 
-	client, err := api.NewDriver(api.ClientConfig{
-		SubscriptionID:    config.SubscriptionID,
-		TenantID:          config.TenantID,
-		ClientID:          config.ClientID,
-		ClientSecret:      config.ClientSecret,
+	var clientConfig = api.ClientConfig{
+		SubscriptionID: config.SubscriptionID,
+		AzureAuthConfig: azclient.AzureAuthConfig{
+			TenantID:        config.TenantID,
+			AADClientID:     config.ClientID,
+			AADClientSecret: config.ClientSecret,
+		},
 		Location:          config.Location,
 		StorageDriverName: config.StorageDriverName,
 		DebugTraceFlags:   config.DebugTraceFlags,
 		SDKTimeout:        sdkTimeout,
 		MaxCacheAge:       maxCacheAge,
-	})
+	}
+
+	if config.ClientSecret == "" && config.ClientID == "" {
```

Comment:
This code seems to be duplicated between the two ANF drivers.  I hope we can [dry](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) that up and support with unit tests.

Reply:
Agree. However, there are many other duplications between `NASBlockStorageDriver` and `NASStorageDriver` not only this code snippet, even the whole `initializeAzureSDKClient` method is almost the same between `NASBlockStorageDriver` and `NASStorageDriver`. So, I think it's better to eliminate the duplication totally in other separate PR, which is an enhancement, and keep the implementation in this PR consistent with the current.